frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -155,26 +162,39 @@ export const getDeploymentData = (resource: K8sResourceKind) => {
     triggers: { image: true, config: true },
   };
   const containers = _.get(resource, 'spec.template.spec.containers', []);
-  if (getResourcesType(resource) === Resources.KnativeService) {
-    deploymentData = {
-      ...deploymentData,
-      env: containers[0]?.env || [],
-      triggers: {
-        image: containers[0]?.imagePullPolicy === ImagePullPolicy.Always,
-      },
-    };
-  } else {
-    const triggers = _.get(resource, 'spec.triggers');
-    deploymentData = {
-      env: _.get(containers[0], 'env', []),
-      triggers: {
-        image: checkIfTriggerExists(triggers, 'ImageChange'),
-        config: checkIfTriggerExists(triggers, 'ConfigChange'),
-      },
-      replicas: _.get(resource, 'spec.replicas', 1),
-    };
+  switch (getResourcesType(resource)) {
+    case Resources.KnativeService: {
+      return (deploymentData = {
+        ...deploymentData,
+        env: containers[0]?.env || [],
+        triggers: {
+          image: containers[0]?.imagePullPolicy === ImagePullPolicy.Always,
+        },
+      });
+    }
+    case Resources.OpenShift: {
+      const triggers = _.get(resource, 'spec.triggers');
+      return (deploymentData = {
+        env: _.get(containers[0], 'env', []),
+        triggers: {
+          image: checkIfTriggerExists(triggers, 'ImageChange', resource.kind),
+          config: checkIfTriggerExists(triggers, 'ConfigChange'),
+        },
+        replicas: _.get(resource, 'spec.replicas', 1),
+      });
```

Comment:
```suggestion
      deploymentData = {
        env: _.get(containers[0], 'env', []),
        triggers: {
          image: checkIfTriggerExists(triggers, 'ImageChange', resource.kind),
          config: checkIfTriggerExists(triggers, 'ConfigChange'),
        },
        replicas: _.get(resource, 'spec.replicas', 1),
      };
      break;
```

Reply:
No need of using break since I am returning the data from here directly.